lib/batchVersions.js

```diff
@@ -249,12 +273,18 @@ export async function insertOptions({options}) {
   return record;
 }
 
-async function _getNextVersionId() {
+export async function _getNextVersionId({explain = false} = {}) {
   const projection = {_id: 0, 'batchVersion.id': 1};
   const sort = {'batchVersion.id': -1};
   const collection = database.collections['tokenization-batchVersion'];
+
+  if(explain) {
+    const cursor = await collection.find({}, {projection, sort}).limit(1);
+    return cursor.explain('executionStats');
+  }
+
   const records = await collection.find(
-    {}, {projection, sort}).toArray();
+    {}, {projection, sort}).limit(1).toArray();
```

Comment:
`limit(1)` is added here to ensure that we only examine 1 index key. If `limit(1)` is not added, it will examine each `batchVersion.id` key for every record in the database.

Reply:
Yeah, we should have had that `limit(1)` there from the start.